src/DependencyInjection/Compiler/KernelPass.php

```diff
@@ -20,11 +20,21 @@ public function process(ContainerBuilder $container)
         foreach ($container->getDefinitions() as $id => $definition) {
             if ($this->isSignalSlot($id) ||
                 $this->isSmartCacheListener($id) ||
-                $this->isResponseCacheListener($id)
+                $this->isResponseCacheListener($id) ||
+                $this->isCachePurger($id)
             ) {
                 $container->removeDefinition($id);
             }
         }
+        $container->removeAlias('ezpublish.http_cache.purger');
+        $arguments = $container->getDefinition('cache_clearer')->getArguments();
+        $arguments[0] =  array_values(array_filter($arguments[0], function ($argument) {
```

Comment:
In Symfony 3.4 `cache_clearer` [service definition changed](https://github.com/symfony/symfony/commit/fa62e5068e81c65cd2676675da4caeec438a33cc#diff-9301964a7acd200e8134fed55aede310R43) and now `$arguments[0]` is an instance of `Symfony\Component\DependencyInjection\Argument\TaggedIteratorArgument`
so during cache warmup we get:
```
In KernelPass.php line 36:
                                                                         
  Warning: array_filter() expects parameter 1 to be array, object given
```
Some nasty `if` checking whether this is an array could solve this, but I have a feeling there's bigger issue here. I can't see direct reference to this change in Symfony upgrade notes, so maybe we're not supposed to "mess" with this service like that?

Reply:
I like how it now removes the tagged services automatically if the service definition is removed.  But in older kernels this is not the case and the tagged service is collected and added to `cache_clearers` by the framework bundle before this compiler pass runs.  Not sure how else to handle this.

Reply:
In worst case detect symfony version with https://github.com/symfony/symfony/blob/master/src/Symfony/Component/HttpKernel/Kernel.php#L67

Reply:
How about a simple `is_array`?  https://github.com/ezsystems/ezplatform-http-cache/pull/17